### ISSUE TITLE
onStateChange: “diff is not defined”

### ISSUE THREAD

**Issue by jondashkyle** [yoshuawuyts/choo-log#2]
Yo this is awesome! Quick note that I’m getting this error when configured as instructed in the examples:

``` javascript
Line 89: Uncaught ReferenceError: diff is not defined
```

Can resolve by updating line 89 to be:

``` javascript
const diff = deepDiff(prev, state) || [ ]
```

Unsure if the root of this one is a fault in how I'm updating the state w/ Choo, or if it’s something to do with the logger itself, so holding off on doing a pull request, etc…

**Comment by yoshuawuyts** [comment]
@jondashkyle oh, thanks for reporting! - reckon like you said it could be because sometimes there is no diff; which is totally feasible I reckon though it might indeed signal that unnecessary updates are happening when they shouldn't - why update if nothing changed?.

Might perhaps be interesting to warn on these updates without a diff; perhaps display `0 diffs` in red or smth.

Think you're on the right track with your fix tho - PR def welcome!

**Comment by yoshuawuyts** [comment]
Closed by https://github.com/yoshuawuyts/choo-log/pull/3 :sparkles: Published as `v1.4.0`
